types/test/findAll.ts

```diff
@@ -0,0 +1,24 @@
+import { expectTypeOf } from 'expect-type'
+import { Sequelize } from 'sequelize';
+import { User } from './models/User';
+
+async () => {
+    const users = await User.findAll({ raw: false });
+    expectTypeOf(users).toEqualTypeOf<User[]>()
+
+    interface RawUser {
+        foo: 'bar';
+    }
+    const rawUsers = await User.findAll<User, RawUser>({
+        attributes: [
+            ['bar', 'foo'],
+            'ignored',
+            [Sequelize.col('table.id'), 'xyz'],
+        ],
+        raw: true,
+    });
+    expectTypeOf(rawUsers).toEqualTypeOf<RawUser[]>();
+
+    // @ts-expect-error
+    rawUsers[0].id = 123; // not an instance
```

Comment:
can you also include tests to make sure that TS infers these generics?

`User.findAll({ raw: true })` should at least infer that `M` is `typeof User`, but I'm not sure about the second generic.

Reply:
What would those tests look like?

In the `findAll` definition, `M` is actually `User` not `typeof User`. The usage of the _instance_ as the type does seem a bit weird to me, but that is existing Sequelize behavior that I would prefer to leave unchanged, so that it is not a breaking change

Reply:
I mean to just check that it works without specifying generics.  It can be a bit confusing for ppl just starting to use TS to get strange behavior with generics.

Reply:
@JacobLey I guess he means like so:
```suggestion
    const users = await User.findAll({ raw: false });
    expectTypeOf(users).toEqualTypeOf<User[]>()

    const users = await User.findAll({ raw: true });
    expectTypeOf(users).toEqualTypeOf<User['_attributes'][]>()

    interface RawUser {
        foo: 'bar';
    }
    const rawUsers = await User.findAll<User, RawUser>({
        attributes: [
            ['bar', 'foo'],
            'ignored',
            [Sequelize.col('table.id'), 'xyz'],
        ],
        raw: true,
    });
    expectTypeOf(rawUsers).toEqualTypeOf<RawUser[]>();

    // @ts-expect-error
    rawUsers[0].id = 123; // not an instance
```